### PR TITLE
compatible with ontology

### PR DESCRIPTION
**Pull request by blockchain-develop**

### CODE REVIEW

File: cmd/utils/ont.go [hunk 1]
```diff
@@ -305,7 +305,12 @@ func SignTransaction(signer *account.Account, tx *types.MutableTransaction) erro
 		tx.Payer = signer.Address
 	}
 	txHash := tx.Hash()
-	sigData, err := Sign(txHash.ToArray(), signer)
+	sink := common.NewZeroCopySink(nil)
+	sink.WriteBytes(txHash.ToArray())
```
Comment (laizy) [inline cmd/utils/ont.go]: add function` Transaction.SigHashForChain(chainId)`  to avoid duplication.

File: cmd/sigsvr/handlers/sig_raw_tx.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func SigRawTransaction(req *clisvrcom.CliRpcRequest, resp *clisvrcom.CliRpcRespo
 		mutable.Payer = signer.Address
 	}
 
-	txHash := mutable.Hash()
+	txHash := tmpTx.SigHashForChain(uint32(1))
```
Comment (laizy) [inline cmd/sigsvr/handlers/sig_raw_tx.go]: avoid hardcoding system id

File: cmd/utils/ont.go [hunk 3]
```diff
@@ -357,7 +361,11 @@ func MultiSigTransaction(mutTx *types.MutableTransaction, m uint16, pubKeys []ke
 		mutTx.Sigs = make([]types.Sig, 0)
 	}
 
-	txHash := mutTx.Hash()
+	txTemp, err := mutTx.IntoImmutable()
+	if err != nil {
+		return err
+	}
+	txHash := txTemp.SigHashForChain(uint32(1))
```
Comment (laizy) [inline cmd/utils/ont.go]: ditto.

Comment (blockchain-develop) [reply cmd/utils/ont.go]: have done

File: cmd/sigsvr/handlers/sig_raw_tx.go (hunk 2 above)
Comment (blockchain-develop) [reply cmd/sigsvr/handlers/sig_raw_tx.go]: have done

File: cmd/utils/ont.go (hunk 1 above)
Comment (blockchain-develop) [reply cmd/utils/ont.go]: have done

**Review by laizy — APPROVED** [review]

**Pull request merged**